dcos/cosmos.py

```diff
@@ -0,0 +1,396 @@
+            'package/render': _data(['v1'], 'post'),
+            'package/repository/add': _data(['v1'], 'post'),
+            'package/repository/delete': _data(['v1'], 'post'),
+            'package/repository/list': _data(['v1'], 'post'),
+            'package/search': _data(['v1'], 'post'),
+            'package/uninstall': _data(['v1'], 'post'),
+            'service/start': _data(['v1'], 'post')
+        }
+
+        self._special_content_types = {
+            ('package/add', 'v1'):
+                'application/vnd.dcos.universe.package+zip;version=v1',
+            ('capabilities', 'v1'):
+                _format_media_type('capabilities', 'v1', '')
+        }
+
+        self._special_accepts = {
+            ('capabilities', 'v1'):
+                _format_media_type('capabilities', 'v1', '')
+        }
+
+    def enabled(self):
+        """
+        Returns whether or not cosmos is enabled on specified dcos cluster
+
+        :return: true if cosmos is enabled, false otherwise
+        :rtype: bool
+        """
+        try:
+            response = self.call_endpoint(
+                'capabilities')
+        # return `Authentication failed` error messages
+        except DCOSAuthenticationException:
+            raise
+        # Authorization errors mean endpoint exists, and user could be
+        # authorized for the command specified, not this endpoint
+        except DCOSAuthorizationException:
+            return True
+        # allow exception through so we can show user actual http exception
+        # except 404, because then the url is fine, just not cosmos enabled
+        except DCOSHTTPException as e:
+            logger.exception(e)
+            return e.status() != 404
+        except Exception as e:
+            logger.exception(e)
+            return True
+        return response.status_code == 200
+
+    def call_endpoint(self,
+                      endpoint,
+                      headers=None,
+                      data=None,
+                      json=None,
+                      **kwargs):
+        """
+        Gets the Response object returned by comos at endpoint
+
+        :param endpoint: a cosmos endpoint, of the form 'x/y',
+        for example 'package/repo/add' or 'service/start'
+        :type endpoint: str
+        :param headers: these header values will appear
+        in the request headers.
+        :type headers: None | dict[str, str]
+        :param data: the request's body
+        :type data: dict | bytes | file-like object
+        :param json: JSON request body
+        :type json: dict
+        :param kwargs: Additional arguments to requests.request
+                       (see py:func:`request`)
+        :type kwargs: dict
+        :return: the Response object returned by cosmos
+        :rtype: requests.Response
+        """
+        if not self._endpoint_exists(endpoint):
+            raise DCOSException(
+                'Cosmos called with unexpected endpoint {}'.format(endpoint))
```

Comment:
The message is confusing. I'll change it so that it's more apparent that this is a programming error, not a cosmos one.